diff --git "a/C:\\Users\\microck\\AppData\\Local\\Temp\\rn-codegen-pr-patch\\orig\\scripts\\codegen\\generate-artifacts-executor\\generateRCTThirdPartyComponents.js" "b/C:\\Users\\microck\\AppData\\Local\\Temp\\rn-codegen-pr-patch\\patched\\scripts\\codegen\\generate-artifacts-executor\\generateRCTThirdPartyComponents.js"
index 1bbfd76..95f3ec7 100644
--- "a/C:\\Users\\microck\\AppData\\Local\\Temp\\rn-codegen-pr-patch\\orig\\scripts\\codegen\\generate-artifacts-executor\\generateRCTThirdPartyComponents.js"
+++ "b/C:\\Users\\microck\\AppData\\Local\\Temp\\rn-codegen-pr-patch\\patched\\scripts\\codegen\\generate-artifacts-executor\\generateRCTThirdPartyComponents.js"
@@ -139,7 +139,7 @@ function generateRCTThirdPartyComponents(
     .flatMap(library => {
       const components = componentsInLibraries[library];
       return components.map(({componentName, className}) => {
-        return `\t\t@"${componentName}": NSClassFromString(@"${className}"), // ${library}`;
+        return `\t\tif (Class component = NSClassFromString(@"\${className}")) {\n\t\t\tcomponents[@"\${componentName}"] = component;\n\t\t} // \${library}`;
       });
     })
     .join('\n');
diff --git "a/C:\\Users\\microck\\AppData\\Local\\Temp\\rn-codegen-pr-patch\\orig\\scripts\\codegen\\templates\\RCTThirdPartyComponentsProviderMM.template" "b/C:\\Users\\microck\\AppData\\Local\\Temp\\rn-codegen-pr-patch\\patched\\scripts\\codegen\\templates\\RCTThirdPartyComponentsProviderMM.template"
index 4ce5107..702537e 100644
--- "a/C:\\Users\\microck\\AppData\\Local\\Temp\\rn-codegen-pr-patch\\orig\\scripts\\codegen\\templates\\RCTThirdPartyComponentsProviderMM.template"
+++ "b/C:\\Users\\microck\\AppData\\Local\\Temp\\rn-codegen-pr-patch\\patched\\scripts\\codegen\\templates\\RCTThirdPartyComponentsProviderMM.template"
@@ -19,9 +19,10 @@
   static dispatch_once_t nativeComponentsToken;
 
   dispatch_once(&nativeComponentsToken, ^{
-    thirdPartyComponents = @{
+    NSMutableDictionary<NSString *, Class<RCTComponentViewProtocol>> *components =
+        [NSMutableDictionary dictionary];
 {thirdPartyComponentsMapping}
-    };
+    thirdPartyComponents = [components copy];
   });
 
   return thirdPartyComponents;
